lib/beaker/shared/subcommands_util.rb

```diff
@@ -30,7 +38,72 @@ def reset_argv(args)
       # @param [Array<String>] args the arguments determined by a specific subcommand
       def execute_beaker(*args)
         reset_argv(args)
-        Beaker::CLI.new.execute! 
+        Beaker::CLI.new.execute!
+      end
+
+      # Checks for the presence of a rake file, and creates a rake file if no rake file is present
+      # @return [String] the name of the rakefile to use
+      def determine_rake_file()
+        RAKE_FILES.each do |rake_file|
+          return rake_file if File.file?(rake_file)
+        end
+        File.open RAKE_FILES.first, "w"
```

Comment:
@ferglor I'm curious as to why you open the file before returning it from this method.

It looks like opening the file [write-only](https://ruby-doc.org/core-2.0.0/IO.html#method-c-new-label-IO+Open+Mode) will truncate the file if it already exists. It looks like the `require_tasks` method does some appending if the file doesn't have our line already, but I worry about this truncation making that code irrelevant. Would you either point out the (or add a) test that confirms that if a Rakefile already exists, you're appending to that file rather than truncating it?

Another point organizationally that I think would be helpful is if we had all of our actual Rakefile manipulations (opening, appending, reading) in the same place if possible. That would make me think that maybe this method should just figure out which filename should be used, & `require_tasks` should handle all manipulations on this file.